crates/biome_js_analyze/src/lint/complexity/use_literal_keys.rs

```diff
@@ -161,3 +172,24 @@ impl Rule for UseLiteralKeys {
 declare_node_union! {
     pub AnyJsMember = AnyJsComputedMember | JsComputedMemberName
 }
+
+fn has_unescaped_new_line(text: &str) -> bool {
+    let mut in_escape_mode = false;
+    for c in text.chars() {
+        match c {
+            '\\' => {
+                in_escape_mode = !in_escape_mode;
+                continue;
+            }
+            '\n' => {
+                if !in_escape_mode {
+                    return true;
+                }
+            }
+            _ => {
+                in_escape_mode = false;
+            }
+        }
+    }
+    false
```

Comment:
This code could be improved by iterating over bytes (this is more efficient) and directly using the iterator to skip the escaped item. Note that it is safe to skip a part of a single byte of a multi-byte character.

```suggestion
    let mut iter = text.as_bytes().iter();
    while let(c) = iter.next() {
        match c {
            b'\\' => {
                iter.next();
            }
            b'\n' => {
                    return true;
            }
            _ => {}
        }
    }
    false
```

Reply:
TIL, thanks!